src/python/pants/backend/python/tasks/python_tool_prep_base.py

```diff
@@ -27,6 +28,17 @@ def __init__(self, pex_path, interpreter):
   def pex(self):
     return self._pex
 
+  def output(self, args, input=None, **kwargs):
+    cmdline = safe_shlex_join(self._pex.cmdline(args))
+    process = self._pex.run(args,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            with_chroot=False,
+                            blocking=False,
+                            **kwargs)
+    (stdout, stderr) = process.communicate(input=input)
+    return (stdout, stderr, process.returncode, cmdline)
```

Comment:
This is going to return bytes in Py3. Are we sure we want that? If not, change to
```suggestion
    return (stdout.decode('utf-8'), stderr.decode('utf-8'), process.returncode, cmdline)
```

Reply:
I was thinking that getting bytes out of a process invocation might be desired by some task (e.g. binary thrift objects -- I'm not referring to any specific task here), and I didn't want to enforce that the output must be unicode.

Reply:
Okay that sounds reasonable and is fine by me! Thanks.

Reply:
(this could potentially be turned into a `binary_mode=False` argument though -- will do that now)